### ISSUE TITLE
app fail to start with inconsistent lifecycle object

### ISSUE THREAD

**Issue by nusnewob** [cloudfoundry/cloud_controller_ng#1275]
Thanks for submitting an issue to `cloud_controller_ng`. We are always trying to improve! To help us, please fill out the following template.

## Issue

V3 API allows assigning docker droplets to app with buildpack lifecycle, then fails to start the app with confusing error.

## Context

```
$ cf v3-create-app docker-sample
$ cf v3-create-package docker-sample --docker-image docker-sample:dev
$ cf curl /v3/builds -X POST -d '{"package":{"guid":"868b4a6c-e0b7-4086-814a-181c91f8bbd7"},"lifecycle":{"type":"docker","data":{}}}'
$ cf curl /v3/apps/0e7d127c-f0c8-467d-b871-a86b7fa664d8/relationships/current_droplet -X PATCH -d '{"data":{"guid":"e72d3251-d81b-449d-aaa2-a3bcf19b1467"}}'
$ cf curl /v3/apps/0e7d127c-f0c8-467d-b871-a86b7fa664d8/actions/start -X POST
{
  "errors": [
    {
      "detail": "Runner invalid request: Invalid field: setup, Invalid field: from, Invalid field: checksum value",
      "title": "CF-RunnerInvalidRequest",
      "code": 170014
    }
  ]
}
```

## Steps to Reproduce

See above.

## Expected result

Sensible error while assigning docker droplets to buildpack based app.

## Current result

```
{
  "errors": [
    {
      "detail": "Runner invalid request: Invalid field: setup, Invalid field: from, Invalid field: checksum value",
      "title": "CF-RunnerInvalidRequest",
      "code": 170014
    }
  ]
}
```
[Logs](https://gist.github.com/nusnewob/9d1ec1a84f4d4a865167f6fef8a55123).

**Comment by zrob** [comment]
@nusnewob can you try to follow your steps except modify the first command `cf v3-create-app docker-sample --app-type docker` ?

**Comment by nusnewob** [comment]
@zrob Yes, I know this works.

**Comment by Gerg** [comment]
It looks like this issue was fixed by 5ae89f60ea659f032a1797b1f20cf8885b3bdf2a, which first appears in [capi-release 1.85.0](https://github.com/cloudfoundry/capi-release/releases/tag/1.85.0).

I'm going to close this. Feel free to re-open if you are still seeing this.
